app/src/main/java/com/fmsys/snapdrop/JavaScriptInterface.java

```diff
@@ -230,18 +230,67 @@ public static String getSendTextDialogWithPreInsertedString(final String text) {
+                "settingsSvg.appendChild(settingsPath);" +
+                "settingsIconButton.appendChild(settingsSvg);" +
+
+                "let aboutIconButton = document.querySelector('.icon-button[href=\"#about\"]');" +
+                "aboutIconButton.parentElement.insertBefore(settingsIconButton, aboutIconButton.nextSibling);" +
+            
+                //remove donation button (play guidelines)
+                "document.querySelector('.icon-button[href*=\"paypal\"]').hidden = true;" +
+            
+                //remove "safari hack"                 
+                "document.body.onclick = null;" +
+            
+                //change ServerConnection.send(message) to connect to JavaScriptInterface
+                "ServerConnection.prototype.s = ServerConnection.prototype.send;" +
+                "ServerConnection.prototype.send = function(message){" +
+                "               this.s(message);" +
+                "               if (message.type == 'pong') {" +
+                "                   SnapdropAndroid.updateLastOnlineTime();" +
+                "               }" +
+                "            };" +
+                
+                //change PeerUI._onTouchEnd(e) to connect to JavaScriptInterface
+                "PeerUI.prototype._oTE = PeerUI.prototype._onTouchEnd;" +
+                "PeerUI.prototype._onTouchEnd = function(e){" +
+                "               this._oTE(e);" +
+                "               if ((Date.now() - this._touchStart < 500) && SnapdropAndroid.shouldOpenSendTextDialog()) {" +
+                "                   Events.fire('text-recipient', e.detail);" +
```

Comment:
Somehow, the send text dialog opens, but sending doesn't work when coming from shortclick. Need to dive into it

Reply:
Ah i see, thats strange....

Reply:
Oh i think i was tired as well, of course it needs to be `this._peer.id` its fixed now